contracts/VestingManager.sol

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+
+import 'hardhat/console.sol';
+
+contract VestingManager is Ownable {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable oidToken;
+    uint256 public referenceDate = 0;
+    uint32 public constant QUARTER = 13 weeks;
+    uint8 public schemaId = 0;
+
+    struct VestingSchema {
+        uint8 lockup; // Number of quarters during which the holders tokens are locked and not accountable for vesting
+        uint16 vesting; // % of total tokens released every quarter with two decimals of resolution (i.e. 1256 -> 12,56%)
+    }
+
+    struct Holding {
+        uint256 totalVestedAmount;
+        uint256 releasedAmount;
+        uint8 vestingSchema;
+    }
+
+    /* Mappings */
+    mapping(uint8 => VestingSchema) public VestingSchemas;
+    mapping(address => Holding[]) public Holdings;
+
+    /* Events */
+    event Deposit(address to, uint256 amount, uint8 schemaId);
+    event Withdraw(address to, uint256 amount);
+    event SchemaCreated(uint8 schemaId, uint8 lockup, uint16 vesting);
+    event EthDeposit(address sender, uint256 value, uint256 balance);
```

Comment:
no one should deposit ethers on the contract, i don't think we need a fallback function

Reply:
Nobody should, not fully sure what happens if someone actually sends some without having the fallback function then. Fallback + withdraw was added mainly, so we have a way to recover ETH in case of wrong deposits